app/views/ai/HackstackAutoPromotion.vue

```diff
@@ -0,0 +1,31 @@
+<script>
+import ModalDynamicPromotion from 'ozaria/site/components/teacher-dashboard/modals/ModalDynamicContent'
+import HackstackPromotionComponent from './HackstackPromotionComponent'
+export default {
+  components: {
+    ModalDynamicPromotion,
+    HackstackPromotionComponent,
+  },
+  methods: {
+    openModalDynamicPromotion () {
+      this.modalDynamicPromotion = true
+    },
```

Comment:
and we could call the modal's onClose like this:
```suggestion
    onClose () {
      this.$refs.modal.onClose()
    },
```